core/modules/catalog/helpers/productOptions/getSelectedOption.ts

```diff
@@ -8,7 +8,7 @@ export default function getSelectedOption (selectedVariant, attributeCode, optio
     value: selectedVariant[attributeCode]
   }
   if (option.values && option.values.length) {
-    const selectedOptionMeta = option.values.find(ov => ov.value_index === selectedOption.value)
+    const selectedOptionMeta = option.values.find(ov => ov.value_index.toString() === selectedOption.value.toString())
```

Comment:
please change it to
`const selectedOptionMeta = option.values.find(ov => String(ov.value_index) === String(selectedOption.value))`
`.toString()` will fail on `null` and `undefined`

Reply:
done

Reply:
there is `.toString()` at the end of line :) Can you remove it?
```
const selectedOptionMeta = option.values.find(ov => String(ov.value_index) === String(selectedOption.value)).toString()
```

Reply:
ahhh.. lol... i thought above was line break... now i get what u wrote before... 🔢

Reply:
done